File: C++/BASIC MATHS REVISION/palindrome_number.cpp
#include <iostream>
using namespace std;
int main(){
    int n;
    cin>>n;
    int revNum=0;
    int dup=n;
    while(n>0){
        int digit = n % 10; // Get the last digit
        revNum = revNum * 10 + digit; // Build the reversed number
        n= n / 10; // Remove the last digit
    }
    cout << "Reversed Number: " << revNum << endl;
    if(dup == revNum){
        cout << "The number is a palindrome." << endl;
    } else {
        cout << "The number is not a palindrome." << endl;
    }
    return 0;
}

File: JAVA/BASIC MATHS REVISION/palindrome_number.java
import java.util.Scanner;

public class palindrome_number {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int revNum = 0;
        int dup = n;

        while (n > 0) {
            int digit = n % 10;
            revNum = revNum * 10 + digit;
            n = n / 10;
        }

        System.out.println("Reversed Number: " + revNum);
        if (dup == revNum) {
            System.out.println("The number is a palindrome.");
        } else {
            System.out.println("The number is not a palindrome.");
        }
    }
}
